src/robot/variables/replacer.py

```diff
@@ -139,31 +139,31 @@ def _get_variable_item(self, match, value):
         for item in match.items:
             if is_dict_like(value):
                 value = self._get_dict_variable_item(name, value, item)
-            elif is_list_like(value):
-                value = self._get_list_variable_item(name, value, item)
+            elif is_iterable(value):
+                value = self._get_iterable_variable_item(name, value, item)
             else:
                 raise VariableError(
-                    "Variable '%s' is %s, not list or dictionary, and thus "
+                    "Variable '%s' is %s, which is not iterable, and thus "
```

Comment:
This error message could be enhanced a little. We explicitly support dictionaries and iterables, and while dicts actually are iterable, in this context they are accessed by key, not by index like iterables. Something like `not a dictionary or iterable` or perhaps `does not support item access` might be better. If you don't come up with something that you like, just leave it and I can think about it later.

Reply:
Updated the error message in f47f4a582